### PR TITLE
Added FacebookVideo entity and admin

### PR DESCRIPTION
**Pull request by rpg600**

### CODE REVIEW

File: app/migrations/Version20170412172114.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
```
Comment (tgalopin) [inline app/migrations/Version20170412172114.php]: Can you remove these comments?

File: app/migrations/Version20170412172114.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+class Version20170412172114 extends AbstractMigration
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
```
Comment (tgalopin) [inline app/migrations/Version20170412172114.php]: Can you remove these statements?

File: app/migrations/Version20170412172114.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+class Version20170412172114 extends AbstractMigration
+{
+    /**
+     * @param Schema $schema
```
Comment (tgalopin) [inline app/migrations/Version20170412172114.php]: Same here

File: app/migrations/Version20170412172114.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        $this->addSql('CREATE TABLE facebook_videos (id BIGINT AUTO_INCREMENT NOT NULL, url LONGTEXT NOT NULL, description LONGTEXT NOT NULL, author VARCHAR(100) NOT NULL, created_at DATETIME NOT NULL, updated_at DATETIME NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE = InnoDB');
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function down(Schema $schema)
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
```
Comment (tgalopin) [inline app/migrations/Version20170412172114.php]: Same here

File: app/migrations/Version20170412172114.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('CREATE TABLE facebook_videos (id BIGINT AUTO_INCREMENT NOT NULL, url LONGTEXT NOT NULL, description LONGTEXT NOT NULL, author VARCHAR(100) NOT NULL, created_at DATETIME NOT NULL, updated_at DATETIME NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE = InnoDB');
+    }
+
+    /**
+     * @param Schema $schema
```
Comment (tgalopin) [inline app/migrations/Version20170412172114.php]: Same here

File: app/migrations/Version20170412172114.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
```
Comment (mykiwi) [inline app/migrations/Version20170412172114.php]: remove comments + abortIf

File: src/AppBundle/Admin/FacebookVideoAdmin.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+{
+    protected $datagridValues = [
+        '_sort_order' => 'DESC',
+        '_sort_by' => 'createAt',
+    ];
+
+    protected function configureFormFields(FormMapper $formMapper)
+    {
+        $formMapper
+            ->add('url', null, [
```
Comment (mykiwi) [inline src/AppBundle/Admin/FacebookVideoAdmin.php]: What about put an UrlType ?

File: src/AppBundle/Admin/FacebookVideoAdmin.php [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                'label' => 'Url',
+            ])
+            ->add('description', null, [
```
Comment (mykiwi) [inline src/AppBundle/Admin/FacebookVideoAdmin.php]: TextareaType maybe ?

File: src/AppBundle/Admin/FacebookVideoAdmin.php (hunk 7 above)
Comment (rpg600) [inline src/AppBundle/Admin/FacebookVideoAdmin.php]: it's guessed

File: src/AppBundle/Admin/FacebookVideoAdmin.php (hunk 8 above)
Comment (rpg600) [inline src/AppBundle/Admin/FacebookVideoAdmin.php]: it's guessed

File: app/Resources/views/page/emmanuel-macron/videos.html.twig [hunk 9]
```diff
@@ -50,70 +50,27 @@
...
-                    {# DEPART : this is the code they should be adding. le reste, c'est nous. #}
-                    <blockquote cite="https://www.facebook.com/EnMarche/videos/769283806583157/" class="fb-xfbml-parse-ignore">
-                        <a href="https://www.facebook.com/EnMarche/videos/769283806583157/"></a>
-                        <p>Laurence Haïm a un message pour vous.
-                           Inscrivez-vous ➜ en-marche.fr/bercy</p>
-                        Posted by <a href="https://www.facebook.com/EnMarche/">En Marche</a> on Tuesday, April 11, 2017
+                    <blockquote cite="{{ video.url }}" class="fb-xfbml-parse-ignore">
+                        <a href="{{ video.url }}"></a>
+                        <p>{{ video.description }}</p>
+                        Publié par {{ video.author }} le {{ video.createdAt|date("m/d/Y", "Europe/Paris") }}
```
Comment (hhamon) [inline app/Resources/views/page/emmanuel-macron/videos.html.twig]: Single quotes in |date() filter.

File: src/AppBundle/Admin/FacebookVideoAdmin.php [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+
+class FacebookVideoAdmin extends AbstractAdmin
+{
+    protected $datagridValues = [
+        '_sort_order' => 'DESC',
+        '_sort_by' => 'createAt',
```
Comment (hhamon) [inline src/AppBundle/Admin/FacebookVideoAdmin.php]: `s/createAt/createdAt/`

File: src/AppBundle/Entity/FacebookVideo.php [hunk 11]
```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity()
```
Comment (hhamon) [inline src/AppBundle/Entity/FacebookVideo.php]: Remove the extra parenthesis if they're not needed.

File: src/AppBundle/Entity/FacebookVideo.php [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+ * @ORM\Table(name="facebook_videos")
+ */
+class FacebookVideo
+{
+    use EntityTimestampableTrait;
+
+    /**
+     * @ORM\Column(type="bigint")
```
Comment (hhamon) [inline src/AppBundle/Entity/FacebookVideo.php]: Bigint? Any valid reasons why allocating such a type for storing a couple of video records?

File: src/AppBundle/Entity/FacebookVideo.php [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
```
Comment (hhamon) [inline src/AppBundle/Entity/FacebookVideo.php]: No need for `(strategy="AUTO")` as this is the default already.

File: src/AppBundle/Admin/FacebookVideoAdmin.php (hunk 10 above)
Comment (rpg600) [inline src/AppBundle/Admin/FacebookVideoAdmin.php]: good catch

File: src/AppBundle/Admin/FacebookVideoAdmin.php (hunk 7 above)
Comment (HeahDude) [inline src/AppBundle/Admin/FacebookVideoAdmin.php]: Better be explicit with types.

File: src/AppBundle/Admin/FacebookVideoAdmin.php (hunk 8 above)
Comment (HeahDude) [inline src/AppBundle/Admin/FacebookVideoAdmin.php]: Agreed with @mykiwi.

File: app/config/services/admin.xml [hunk 14]
```diff
@@ -66,6 +66,14 @@
             <argument />
         </service>
 
+        <service id="app.admin.facebook_video" class="AppBundle\Admin\FacebookVideoAdmin">
+            <tag name="sonata.admin" manager-type="orm" label="Contenu » Vidéos Facebook" group="Communication" />
+
+            <argument />
```
Comment (HeahDude) [inline app/config/services/admin.xml]: Can you please add comments for empty args?

Comment (rpg600) [inline app/config/services/admin.xml]: These arguments are filled by the sonata admin compiler pass, i think there is no need to add comments, furthermore we will need to update the whole configuration file.

Comment (HeahDude) [inline app/config/services/admin.xml]: Fair enough!

**Pull request merged**